src/test/java/dao/FruitStorageDaoTests.java

```diff
@@ -0,0 +1,167 @@
+package dao;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.util.HashMap;
+import java.util.Map;
+import model.Fruit;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class FruitStorageDaoTests {
+    private FruitStorageDao fruitStorageDao;
+
+    @BeforeEach
+    void setUp() {
+        fruitStorageDao = new FruitStorageDaoImpl();
+
+        fruitStorageDao.set("banana", 100);
+        fruitStorageDao.set("apple", 25);
+    }
+
+    @Test
+    void getFruit_ok() {
+        String fruitName = "banana";
+        int fruitQuantity = 100;
+        Fruit expected = new Fruit(fruitName, fruitQuantity);
+        Fruit actual = fruitStorageDao.getFruit("banana");
+
+        assertEquals(expected.getName(), actual.getName(),
+                "Actual and expected fruit name must be equals");
+
+        assertEquals(expected.getQuantity(), actual.getQuantity(),
+                "Actual and expected fruit q-ty must be equals");
+    }
+
+    @Test
+    void getInvalidFruit_notOk() {
+        String fruitName = "coconut";
+        assertThrows(RuntimeException.class, () -> fruitStorageDao.getFruit(fruitName),
+                "There is no such fruit: " + fruitName);
+    }
+
+    @Test
+    void set_ok() {
+        String fruitName = "banana";
+        int fruitQuantity = 50;
+        Fruit expected = new Fruit(fruitName, fruitQuantity);
+        fruitStorageDao.set("banana", 50);
+        Fruit actual = fruitStorageDao.getFruit("banana");
+
+        assertEquals(expected.getName(), actual.getName(),
+                "Actual and expected fruit names must be equals");
+        assertEquals(expected.getQuantity(), actual.getQuantity(),
+                "Actual and expected fruit q-ty must be equals");
```

Comment:
generate equals and hashcode for fruit, and compare each fruit field at once

Reply:
> generate equals and hashcode for fruit, and compare each fruit field at once

Didn't know I can modify source code. Thx)